frontend/src/api/task.ts

```diff
@@ -0,0 +1,68 @@
+import { post } from "src/api/requests";
+
+/**
+ * Initializes the shape for the data being retrieved from the backend.
+ */
+export interface Task {
+  _id: string;
+  title: string;
+  description?: string;
+  isChecked: boolean;
+  dateCreated: Date;
+}
+
+interface TaskJSON {
+  _id: string;
+  title: string;
+  description?: string;
+  isChecked: boolean;
+  dateCreated: string;
+}
+
+/**
+ * Converts taks from "any"-typed JSON that only contains primitive types to our
+ * custom Task interface. This is used to change the date field from a string
+ * to a Date object.
+ *
+ * @param task the JSON representation of the task
+ * @returns the typed Task
+ */
+function parseTask(task: TaskJSON): Task {
+  return {
+    _id: task._id,
+    title: task.title,
+    description: task.description,
+    isChecked: task.isChecked,
+
+    // convert the string version of the date to a Date object
+    dateCreated: new Date(task.dateCreated),
+  };
+}
+
+/**
+ * This interface is optional, it's being used to make the input of createTask
+ * cleaner. It's being exported so when calls are made to the function it can
+ * be used.
+ */
+export interface CreateTaskRequest {
+  title: string;
+  description?: string;
+}
+
+export async function createTask(task: CreateTaskRequest): Promise<Task> {
+  // we can call "/api/task" due to the "proxy" field in package.json
+  // the proxy field is being used to circumvent CORS errors on the dev system
+  // CORS is a security feature meant to prevent malicious cross-origin
+  // data retrieval
+  //
+  // read more about CORS: https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS
+  // read more about proxying: https://create-react-app.dev/docs/proxying-api-requests-in-development/
```

Comment:
Tbh I think we can leave this information out because for the most part, it's not something TSE developers should have to worry about. We can include it in a further reading section, or as an aside when we summarize the `package.json`

```suggestion
```

Reply:
Personally, I think would be confused if I saw just /api/task without the full address. In fact, I think I was confused by this very thing when I first joined TSE which is why I made the comment. I think it doesn't hurt and isn't an info overload.

Reply:
That's why I'm saying we should briefly overview the `package.json` file and explain each part in the writeup. There we'll have the opportunity to describe the `"proxy"` field, but I still think that the topic of CORS is out of scope. Again, that's something that the EM should set up a process for, so it's irrelevant to developers in TSE and would distract from the purpose of this onboarding repo. Same too-much-information issue as with previous orientation sessions. If a dev is really curious about `"proxy"`, then they can go read the documentation about it and discover for themselves that it also helps prevent CORS errors.